addons/account/static/src/js/account_portal_sidebar.js

```diff
@@ -0,0 +1,51 @@
+odoo.define('account.AccountPortalSidebar', function (require) {
```

Comment:
Can'y you define a basic sidebar in portal and improve the widget here to add stuff in it ? At first sight it looks like a "a bit oriented object, a bit totally hardcoded" methodology.

Reply:
Done